### ISSUE TITLE
Diagnose unqualified calls to std:: functions, especially `move` and `forward`

### ISSUE THREAD

**Issue by Quuxplusone** [llvm/llvm-project#53715]
cc: @cor3ntin

As recently discussed in WG21's LEWG, something like this would be nice:
```
    warning: unqualified name resolves to 'std::move'
        auto y = move(x);
                 ^~~~
    note: use the qualified name 'std::move' instead
        auto y = move(x);
                 ^~~~
                 std::move
```

This should certainly apply to `move` and `forward`, but probably _anything_ in `namespace std` should be considered fair game. If we apply this to _anything_, then we need to have some kind of exception syntax. For example, we want to diagnose
```
std::vector<int> v = {1,2,3};
// This is an actual portability issue! It will break if vector::iterator is a raw pointer type.
sort(v.begin(), v.end());  // expected-warning {{unqualified name resolves to 'std::sort'}}
```
but we don't want to diagnose
```
std::vector<int> v = {1,2,3};
// This is totally fine; it's the "std::swap two-step"
using std::swap;
swap(v, v);  // expected-no-diagnostic
```
We don't even want to diagnose unqualified use of `swap` from _within_ `namespace std`, because the standard library does that all the time on purpose.

**Comment by pdimov** [comment]
If you do this, make sure to make `move` and `forward` a separate warning group, because these have very few false positives and should generally be left enabled, whereas the others, not so much on either count.

In fact, the `move` warning should probably be limited to the one argument `move`.

**Comment by aaronpuchert** [comment]
Has this been solved by 70b1f6de539867353940d3dcb8b25786d5082d63 or is there more to do? The comments suggest this is enough for now and we could close this.
